.github/workflows/build-release.yaml

```diff
@@ -0,0 +1,96 @@
+name: Build and release
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    tags:
+      - "**"
+
+concurrency:
+  group: ${{ github.repository_id }}-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "MacOS x86"
+            runner: macos-12-large
+            release-suffix: macosx-amd64
+          - name: "MacOS arm64"
+            runner: [self-hosted, macOS, ARM64]
+            release-suffix: macosx-arm64
+          - name: "Linux x86"
+            runner: matterlabs-ci-runner
+            image: matterlabs/llvm_runner:ubuntu22-llvm17-latest
```

Comment:
I recall build errors with newer gcc (like the default one in Ubuntu-22.04) when building older versions. IIRC, gcc-9.4 was able to build older versions. If we're using this configuration for all versions, can you confirm if we're able to build all of them? Thanks!

Reply:
Regardless of the newer Docker image, we still set `gcc-9` as the default one:
https://github.com/matter-labs/compiler-infra/blob/main/images/llvm_runner/ubuntu22-llvm17.Dockerfile#L58-L61

so, there should be no difference. But I'll test it thoroughly before merging.

Reply:
I've introduced a new public docker image `matterlabs/llvm_runner:ubuntu20-llvm17-latest` specially for this case.